__tests__/src/extend/withPlugins.test.js

```diff
@@ -19,15 +19,17 @@ function createPluginHoc(pluginMap) {
 }
 
 describe('withPlugins', () => {
-  it('should return a function (normal function call)', () => {
-    expect(withPlugins('Target', Target)).toBeInstanceOf(Function);
+  it('should return a React.memo object (normal function call)', () => {
+    expect(withPlugins('Target', Target)).toBeInstanceOf(Object);
+    expect(withPlugins('Target', Target).type).toBeInstanceOf(Function);
   });
 
-  it('should return a function (curry function call)', () => {
-    expect(withPlugins('Target')(Target)).toBeInstanceOf(Function);
+  it('should return a React.memo object (curry function call)', () => {
+    expect(withPlugins('Target')(Target)).toBeInstanceOf(Object);
+    expect(withPlugins('Target')(Target).type).toBeInstanceOf(Function);
   });
 
-  it('displayName prop of returned function is based on target name argument', () => {
+  xit('displayName prop of returned function is based on target name argument', () => {
```

Comment:
I couldn't figure out how to get this to return properly.

Reply:
There are also other tests failing in withPlugin. I already run into similar issues with that memo function. The problem is that the memo function does not return a component function but a react element object. And this object is somewhat opaque and not described in the docs.

What about putting the PluginHoc at top level and then doing a name export for testing purpose?